src/target/riscv/riscv-013.c

```diff
@@ -587,6 +587,7 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in,
 			break;
 		} else {
 			LOG_ERROR("failed %s at 0x%x, status=%d", op_name, address, status);
+			dtmcontrol_scan(target, DTM_DTMCS_DMIRESET);
 			return ERROR_FAIL;
```

Comment:
Here you can just reset the DMI and break from the loop. The error message and function return is already handled below.

```suggestion
			dtmcontrol_scan(target, DTM_DTMCS_DMIRESET);
			break;
```

Reply:
Done.